internal/internal_workflow_client.go

```diff
@@ -1021,6 +1023,10 @@ func (wc *WorkflowClient) UpdateWorkflowWithOptions(
 	if err := wc.ensureInitialized(); err != nil {
 		return nil, err
 	}
+	// Default update ID
+	if req.UpdateID == "" {
+		req.UpdateID = uuid.New()
```

Comment:
```suggestion
		reqVar := *req
		req = &reqVar
		req.UpdateID = uuid.New()
```

Shallow copy avoids the mutation of user value (if they are updating in a loop for example and pre-made the request, there'd be a problem). I'm sure we don't do this everywhere, but this is why many Go functions do not accept pointer-based options.

I should have brought this up in #989 and guess I didn't. This matches query but not start workflow. I don't have a strong opinion I guess on accepting pointer or non in that case, but we should not mutate the param.